### PR TITLE
[runtime] Initialize MethodBody objects by calling a ctor instead of accessing the object directly. This allows the MethodBody type to be removed by the linker.

### PR DESCRIPTION
**Pull request by vargaz**
Contributes to #9333

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build deb with monolite

**Comment by vargaz** [comment]
@monojenkins build failed

File: mcs/tools/linker/Tests/mscorlib/test-reflection-04.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	public int AProperty {
+		get {
+			return 0;
+		}
+		set {
+		}
+	}
+}
+
+[Foo]
```
Comment (marek-safar) [inline mcs/tools/linker/Tests/mscorlib/test-reflection-04.cs]: Remove this (there is no parameterless ctor)

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**